### ISSUE TITLE
An error occurred while extracting the archive.

### ISSUE THREAD

**Issue by prasha6** [sparkle-project/Sparkle#837]
Hello,

I am integrating sparkle into my app and testing it out. Currently, I am in a debug build ( I am running the app out of my Xamarin IDE). I also have an updated app in a http server. This updated .app is signed using an apple developer id, and it has been compressed into a .zip archive. I have provided the path to this .zip archive in the enclosure tag of my appcast.xml file.

Given this situation, when I run my app, I am expecting it to install the newer version of the app - but I am getting the following errors in my Application output window.

Sparkle: The update archive removes the DSA key, and the old app hasn't been signed with app Code Signing. At least one method of signature verification must be valid. The update will be rejected.
Sparkle: Error: An error occurred while extracting the archive. Please try again later. The update is improperly signed. (URL (null))

Could you please let me know how I could test out the sparkle "Install and relaunch" scenario?

Thanks,
Prashanth

**Comment by pornel** [comment]
All scenarios will fail, because the app isn't properly packaged.

> The update archive removes the DSA key, and the old app hasn't been signed with app Code Signing. At least one method of signature verification must be valid.

You forgot to add DSA key to the new app.

**Comment by prasha6** [comment]
Thanks for getting back pornel.

I actually did add the DSA key. Following is what I did.

For the Updated app:
1. I had the .app signed using apple developer id.
2. I compressed it in a .zip
3. I generated the dsa key using sparkle's ./bin/generate_keys
4. This gave me an output which I added to the sparkle:dsaSignature attribute in the appcast.xml.

For the current app:
5. Step 3 created a dsa_pub.pem file, which I added it to my Resources folder int he current project.
6. I also added a key in my info.plist SUPublicDSAKeyFile with its value as "dsa_pub.pem"

After the above steps, I build and run my current app. This gives me the above error. Are my steps right ?
Thanks a lot for your help!
